packages/common/src/logger-provider.ts

```diff
@@ -46,9 +48,19 @@ export class LoggerProvider {
       this._diagnosticLogger = this._makeDiagnosticLogger()
     }
 
+    private _getLogPath(filename: string) {
+        if (!this.config.logToFiles) {
+            throw new Error("Tried to generate log path when logToFiles is false") // This indicates a programming error
+        }
+        const logDirectory = this.config.logDirectory || path.join(os.homedir(), ".ceramic", "logs/")
```

Comment:
Why couldn't the logger itself take care of setting the path?

Reply:
It could, but I don't think that helps in any way?

Reply:
Doesn't it? The `logDirectory` should only be relevant to the filesystem logger. Am I missing something?

Reply:
So instead of having just the `RotatingFileStream` provided into the logging system from outside (in our case from the `cli` package when setting up the daemon), you want a implementation of the `DiagnosticLogger` and `ServiceLogger` to be provided, and have those handle the config file parsing to set the log directory?  That would work, but is a larger refactoring.

Reply:
Ok, I don't know exactly how the different components fit together. I just intuitively felt that it would make sense for directory to only be set for the loggers that use it. I'm fine with not implementing that now though.